src/App.tsx

```diff
@@ -1,24 +1,207 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoErrors } from './components/TodoErrors';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { FilteredParams } from './types/FilteredParams';
+import { addTodo, deleteTodo, getTodos } from './api/todos';
+import { USER_ID } from './utils/userId';
+import { TodoForm } from './components/TodoForm';
 
-const USER_ID = 0;
+function getPreparedTodos(todosForFilter: Todo[], filterField: FilteredParams) {
+  return todosForFilter.filter(todo => {
+    switch (filterField) {
+      case FilteredParams.active:
+        return !todo.completed;
+
+      case FilteredParams.completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [filter, setFilter] = useState(FilteredParams.all);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loader, setLoader] = useState(true);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [temp, setTemp] = useState<Todo | null>(null);
+
+  const unSetError = () => {
+    setErrorMessage('');
+  };
+
+  // eslint-disable-next-line max-len
+  const preparedTodos = useMemo(() => getPreparedTodos(todos, filter), [todos, filter]);
+
+  const tempMarkup = (
+    <div className="todo">
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+        />
+      </label>
+
+      <span
+        className="todo__title"
+      >
+        {temp?.title}
+      </span>
+
+      <button
+        type="button"
+        className="todo__remove"
+      >
+        ×
+      </button>
+
+      <div className={classNames('modal overlay', {
+        'is-active': temp !== null,
+      })}
+      >
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+    </div>
+  );
+
+  const todosCheck = todos.length > 0;
+
+  const createdNewTodo = async () => {
+    const newTodo = {
+      id: 0,
+      userId: USER_ID,
+      title: todoTitle,
+      completed: false,
+    };
+
+    setTemp(newTodo);
+
+    try {
+      const addedTodo = await addTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch (error) {
+      setErrorMessage('Something went wrong with addition todo');
+      throw error;
+    } finally {
+      setTemp(null);
+    }
+  };
+
+  const removedTodo = async (todoId: number) => {
+    setLoader(true);
```

Comment:
when deleting one todo, the loader appears for all
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/54936044/e11575c5-f8b6-4841-a335-6e14531b14f9)